browser/scripts/collapsible-select-control.js

```diff
@@ -173,6 +173,153 @@ CollapsibleSelectControl.prototype.render = function(el) {
 
 	$('li', el).dblclick(function(e) {
 		that._cb($(e.target).data('path'))
+		$(window).unbind('mousemove')
+	})
+
+	var dragging = false
+	var mouseMoveBound = false
+	var mouseDown = false
+	var mouseX = 0
+	var mouseY = 0
+	var scrollInterval
+	var scrollBound = false
+
+	// Drag and drop an element from the list
+	$('li', el).mousedown(function(e) {
+
+		dragging = true
+		mouseDown = true
+
+		var title = $(e.target).text()
+
+		var dragPreview = $('<div class="plugin-drag-preview"><span style="display: none;">Drop to create:</span>'+title+'</div>')
+		var hoverArea = $('<div class="dragging-allowed"></div>');
+		var dragPreviewInDom = false // only append the preview element when moving the mouse cursor while dragging
+
+		var canvas = $('#canvases')
```

Comment:
just as a note, maybe `var $canvas` could be a better name for this (and dragPreview and hoverArea) since then it's more clearly a jQuery object. IMHO